### PR TITLE
docs: Supply Dockerfile for html docs build.

### LINKED ISSUE #22655: Build the docs using Docker
FWIW, if building just the HTML docs is enough, then the following Dockerfile seems to work:

```dockerfile
FROM python:3.8.12-slim-buster

RUN apt-get update
RUN apt-get install -y make librsvg2-bin imagemagick graphviz git

COPY requirements.txt /tmp
RUN egrep "phinx|mpmath|matplotlib" /tmp/requirements.txt > /tmp/req_min.txt
RUN pip install --upgrade pip
RUN pip install -r /tmp/req_min.txt

CMD cd /sympy/doc; make html
```

If you saved this as `release/Dockerfile.htmldoc`, then you could build the image with

```
$ cd release
$ docker build -f Dockerfile.htmldoc -t sympy_htmldoc .
```

and once you had the image you could build the docs at any time with

```
$ docker run --rm -v /path/to/the/sympy/repo:/sympy sympy_htmldoc
```

The image weighs in at 679MB. If you wanted to also build LaTeX docs I think the image might be 5 to 10 times as large. But I find HTML docs are enough, if I just want to check that things are looking about right.

_Originally posted by @skieffer in https://github.com/sympy/sympy/issues/22629#issuecomment-991698653_

I think it would make more sense to put the Docker file in the doc directory. The instructions for building the docs could be updated to suggest using it as an option.

_Originally posted by @oscarbenjamin in https://github.com/sympy/sympy/issues/22629#issuecomment-991922573_

### PR DESCRIPTION
**Pull request by skieffer**
#### References to other Issues or PRs

Fixes #22655 

#### Brief description of what is fixed or changed

Provides one option to help developers build the HTML version of the docs.

#### Release Notes

NO ENTRY

### CODE REVIEW

File: doc/Dockerfile.htmldoc [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+#
+# The documentation will be built in SYMPY_ROOT's doc/_build/html directory.
+#
+# To view the docs, you can run a simple web server on a port of your choice
+# with, e.g.
+#
+#   $ cd SYMPY_ROOT/doc/_build/html
+#   $ python -m http.server 8888
+#
+# then navigate your browser to http://localhost:8888 (or substitute the port
```
Comment (asmeurer) [inline doc/Dockerfile.htmldoc:26]: It isn't necessary to run a webserver to view the docs. You can just open `index.html`.

**Comment by asmeurer** [comment]
Does this put the built docs inside the image or does it put it in the normal location outside of the image?

File: doc/Dockerfile.htmldoc (hunk 1 above)
Comment (skieffer) [reply doc/Dockerfile.htmldoc:26]: Okay, I'll cut this bit, in order to simplify things.

**Comment by skieffer** [comment]
> Does this put the built docs inside the image or does it put it in the normal location outside of the image?

Outside the image, in the normal location under `doc/_build/html`.

**Comment by oscarbenjamin** [comment]
It would be good to mention this dockerfile somewhere e.g. here:
https://docs.sympy.org/latest/documentation-style-guide.html#installation

**Pull request merged**